lib/Doctrine/DBAL/DBALException.php

```diff
@@ -168,6 +168,9 @@ private static function wrapException(Driver $driver, \Exception $driverEx, $msg
     private static function formatParameters(array $params)
     {
         return '[' . implode(', ', array_map(function ($param) {
+            if (is_string ($param)) {
+                $param = mb_strimwidth($param, 0, 240, "...");
```

Comment:
If you really want to truncate, then then length should also be part of the string here.